plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -400,6 +466,87 @@ protected void copyTableSchema(ConnectorSession session, Connection connection,
         }
     }
 
+    @Override
+    public Optional<JdbcTableHandle> getTableHandle(ConnectorSession session, SchemaTableName schemaTableName)
```

Comment:
I am not sure because of this description from the doc for this method: 
` ResultSetMetaData getMetaData() throws SQLException;`

NOTE: Using this method may be expensive for some drivers due to the lack of underlying DBMS support.

Reply:
So maybe we should provide an option for JDBC connector to decide how table handle should be provided. Depending on the driver (vendor) we may choose different implementation.

Do you see any reason why it should not be a default way for SqlServer?

Reply:
To be honest I don't see any reasons. However do we have some way to understand, that we didn't introduce any regression ?

Reply:
Can we extract some interface that we can inject into `BaseJdbcClient` and we could have two implementations.

Reply:
Possibly. Possibly in future we may have more than 2 if that implementation is tricky and requires more custom things.

Reply:
Looks like there is another way with minimal code changes, please take a look @ssheikin @kokosing